wallet/service/service.go

```diff
@@ -46,19 +46,13 @@ func (service *Service) ReadableDatastore() ReadOnlyDatastore {
 func (service *Service) GetOrCreateWallet(ctx context.Context, walletID uuid.UUID) (*wallet.Info, error) {
 	wallet, err := service.ReadableDatastore().GetWallet(walletID)
 	if err != nil {
-		return nil, errorutils.Wrap(err, "error looking up wallet")
+		return nil, errorutils.Wrap(err, "error looking up wallet in db")
 	}
 
-	if wallet == nil {
-		wallet, err = service.LedgerClient.GetWallet(ctx, walletID)
+	if wallet == nil || wallet.PayoutAddress == nil {
+		wallet, err = service.UpsertWallet(ctx, walletID)
 		if err != nil {
-			return nil, errorutils.Wrap(err, "error looking up wallet")
-		}
-		if wallet != nil {
-			err = service.Datastore.UpsertWallet(wallet)
-			if err != nil {
-				return nil, errorutils.Wrap(err, "error saving wallet")
-			}
+			return nil, err
```

Comment:
It looks like we will upsert the wallet whenever a wallet without an anonymous address is supplied. I'm not sure how often this is called, but if it's called often this might have some performance cost since most wallets don't have addresses.

```
grants_db=> select count(*), payout_address is null from wallets group by 2;
  count   | ?column?
----------+----------
    47140 | f
 17096976 | t
```

Reply:
:heavy_check_mark:
To level set, GetAvailablePromotions always calls GetOrCreateWallet, and the lion share of grant traffic is to that endpoint.